### DISCUSSION TITLE
Stable Environment

### DISCUSSION THREAD

**Discussion by PAkhidue** [gamedaoco/gamedao-haiku#400] [General]
Hey everyone, so we talked on monday about the stable environment and that we definitely need one.

So here we start a discussion to go on with this topic.

There are some open questions like:
- Do we want a new chain (Parachain? we don't want standalone chains anymore?)
- Additional Frontend instance
- Version handling due to Network-Switcher
- Deployment Pipeline for everything about the new environment

Let's start with the new chain: I think we should have an extra chain for our stable environment to provide realistic and clean data to the public. Most of the developers (FE) use the beeblebrox chain for testing and developing features. There are a lot of useless or ugly organizations, proposals and campaigns. The plan was to go with the new release v59. Because of that I think we should have another chain only for the stable environment.

Also we should have an additional Frontend instance. It can happen that broken tickets are getting merged into the main branch, so the whole live product will get redeployed and maybe contains broken logic even if it's only styling etc... We should have a second frontend instance only deploying "releases" that are well-tested so we always have something we can show to the public.

If we have a new instance, including the whole infrastructure (graph) we should set up an deployment pipeline for new releases. Question here is more like: Are we using GitHub or still Gitlab? We should definitely clarify that. 

Another problem is we're having a Network-Switcher. We always need to keep in mind that our GraphQL structure could change. The exintrinsic interfaces could change, also events etc. Therefore we need to find a solution how to work with it. Will we use our "feature-toggle" to enable a specific version for network? So the frontend knows, "I'm connected to a network with version v58, so I need to run this extrinsic, show additional form fields..." and so on. Currently we have 3 networks in the switch but with two different versions. So the problem exists already.

What are your opinions? @2075 @5-mark @vasylenko-yevhen @vovacha @DarkNebula0

**Comment by 2075** [comment]
thanks for the write up @PAkhidue 


1. from my perspective we should move like this:

- use beeblebrox if needed for dev (agree with the ugly bit and that is why I suggested to flush it often)
- use the parachain staging setup to have a standalone relay/para chain and a stable release on it. this should be the chain/fe where which we also use for onboarding of new clients and do our own governance in public

2. also:
- main branch must never break!
- dev branch should never break!
- any other branch can break

3. release / ci / cd
- use release tags to trigger a dedicated ci/cd for network/graph/frontend ( do not believe this is asap possible )
- until then use release tags to identify what to trigger manually. e.g. frontend can be updated more often in stable
- use gitlab, it is proven, known, flexible, also was decided a while ago

4. graph
- the graph needs to be owned in terms of knowing what is going on. we cannot afford anymore unknown unknowns there.

5. network switcher
- we can remove it on the staging to make it cleaner
- the features must be used to scope what features are avaiable
- in the future this could be a mix of discovery via metadata but not now

6. as far as I understood, we now have
- a relay / parachain setup for staging
- a working graph for staging and we know how to confidently update it ( I don't know yet )
- a working frontend not feature par with the chain extrinsics / graph

7. we also need to collect and resolve the diff between product feature request for 1.0 and the staging. does anybody have an overview?
